samples/renderer/src/winmain.cpp

```diff
@@ -0,0 +1,211 @@
+// TODO: consider making this into a separate pacakge that will be later transformed into engine package
+
+#include <chrono>
+#include <memory>
+#include <span>// left to clarify that c++20 is used
+
+#include <rcbe-engine/camera/Camera.hpp>
+#include <rcbe-engine/core/AbstractInputManager.hpp>
+#include <rcbe-engine/core/EditorInputManager.hpp>
+#include <rcbe-engine/core/WindowManager.hpp>
+
+#include <rcbe-engine/datamodel/core/CoreObject.hpp>
+#include <rcbe-engine/datamodel/math/Vector.hpp>
+#include <rcbe-engine/datamodel/rendering/Material.hpp>
+#include <rcbe-engine/datamodel/rendering/RasterizerTexture.hpp>
+#include <rcbe-engine/datamodel/rendering/Shader.hpp>
+#include <rcbe-engine/datamodel/rendering/camera_config.hpp>
+#include <rcbe-engine/datamodel/rendering/rasterizer_texture_types.hpp>
+#include <rcbe-engine/datamodel/rendering/renderer_config.hpp>
+#include <rcbe-engine/datamodel/visual/Texture.hpp>
+#include <rcbe-engine/datamodel/visual/texture_types.hpp>
+#include <rcbe-engine/fundamentals/convinience.hpp>
+#include <rcbe-engine/parsers/gltf/gltf_parser.hpp>
+#include <rcbe-engine/parsers/stl/stl_parser.hpp>
+#include <rcbe-engine/parsers/tga/tga_parser.hpp>
+#include <rcbe-engine/parsers/x3d/x3d_parser.hpp>
+#include <rcbe-engine/renderer/Renderer.hpp>
+#include <rcbe-engine/utils/json_utils.hpp>
+#include <rcbe-engine/utils/output_utils.hpp>
+
+#include <rdmn-engine/logger/trivial_logger.hpp>
+
+int APIENTRY wWinMain(_In_ HINSTANCE hInstance,
+                     _In_opt_ HINSTANCE hPrevInstance,
+                     _In_ LPWSTR    lpCmdLine,
+                     _In_ int       nCmdShow) {
+    UNREFERENCED_PARAMETER(hPrevInstance);
+    UNREFERENCED_PARAMETER(lpCmdLine);
+    using rcbe::core::InputManagerImplementation;
+    try {
+        rcbe::utils::setup_logging();
+        rcbe::core::WindowManager manager {hInstance,/*window_class=*/ L"RDMNWindow", nCmdShow};
+
+        rcbe::core::window_config wconf{};
+        wconf.caption = L"RCBE Engine";
+        wconf.background_color = rcbe::visual::RGBAColor(0.5, 0.5, 0.5, 1.0);
+        wconf.position = decltype(wconf.position)(10.0, 10.0);
+        wconf.size = rcbe::core::IntegralDimensions(1024, 768);
+        wconf.process_input = true;
+        wconf.editor = true;
```

Comment:
Save these config to json file, and read from that file.

Reply:
I'll leave a `TODO`, right now there are some issues with relation to `wstring` window caption.

Reply:
Actually did refactored to config read.